flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -307,9 +306,10 @@ public void dropTempCatalogFunction(ObjectIdentifier identifier, boolean ignoreI
 			);
 		}
 
-		Catalog catalog = catalogManager.getCatalog(oi.getCatalogName()).get();
+		Optional<Catalog> catalogOptional = catalogManager.getCatalog(oi.getCatalogName());
 
-		if (catalog != null) {
+		if (catalogOptional.isPresent()) {
```

Comment:
nit: This could be simplified further:

```
return catalogOptional
  .map(...);
```

I do understand it is independent of the PR.